hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -18,38 +17,38 @@
 
 package org.apache.hudi.keygen;
 
-import org.apache.hudi.DataSourceWriteOptions;
-import org.apache.hudi.common.config.TypedProperties;
-
 import org.apache.avro.generic.GenericRecord;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
+import org.apache.spark.sql.Row;
 
 import java.util.Arrays;
 import java.util.stream.Collectors;
-import org.apache.spark.sql.Row;
 
 /**
  * Complex key generator, which takes names of fields to be used for recordKey and partitionPath as configs.
  */
 public class ComplexKeyGenerator extends BuiltinKeyGenerator {
 
-  public static final String DEFAULT_RECORD_KEY_SEPARATOR = ":";
+  private CommonComplexKeyGenerator commonComplexKeyGenerator;
 
   public ComplexKeyGenerator(TypedProperties props) {
     super(props);
-    this.recordKeyFields = Arrays.stream(props.getString(DataSourceWriteOptions.RECORDKEY_FIELD_OPT_KEY())
+    this.recordKeyFields = Arrays.stream(props.getString(KeyGeneratorOptions.RECORDKEY_FIELD_OPT_KEY)
```

Comment:
what the user sets is is actually `DataSourceWriteOptions.RECORDKEY_FIELD_OPT_KEY()`. So might be good to avoid duplicating this in `KeyGeneratorOptions.RECORDKEY_FIELD_OPT_KEY` . can we revert?

Reply:
> what the user sets is is actually `DataSourceWriteOptions.RECORDKEY_FIELD_OPT_KEY()`. So might be good to avoid duplicating this in `KeyGeneratorOptions.RECORDKEY_FIELD_OPT_KEY` . can we revert?

`DataSourceWriteOptions` has some `hive` and `spark` quotes in it.  it is a little inconvenient to move to `hudi-client-common`. should I refactor `DataSourceWriteOptions` ?

Reply:
ah ok. no problem. this makes sense